data/deduplicator/deduplicator/device_deactivate_hash.go

```diff
@@ -5,18 +5,17 @@ import (
 
 	"github.com/tidepool-org/platform/data"
 	dataStore "github.com/tidepool-org/platform/data/store"
+	"github.com/tidepool-org/platform/data/types"
 	dataTypesUpload "github.com/tidepool-org/platform/data/types/upload"
 	"github.com/tidepool-org/platform/errors"
 	"github.com/tidepool-org/platform/page"
 	"github.com/tidepool-org/platform/pointer"
 )
 
-type DeviceDeactivateHashVersion string
-
 const (
-	DeviceDeactivateHashVersionUnkown  DeviceDeactivateHashVersion = ""
-	DeviceDeactivateHashVersionCurrent DeviceDeactivateHashVersion = "1.1.0"
-	DeviceDeactivateHashVersionLegacy  DeviceDeactivateHashVersion = "0.0.0"
+	DeviceDeactivateHashVersionUnknown = "unknown-hash"
```

Comment:
Sorry, I think we have a disconnect. I may not have explained myself very well. These version strings need to remain the same.

There need to be two separate version groups. 

First, the version of the DeviceDeactivationHash deduplicator. That would be "0.0.0" (for Jellyfish) and "1.1.0" (for current Platform).

Second, the version of the identity fields, that would be, for simplicity, "0" (for legacy) and "1" (for current). This is what is passed into the datum.IdentityFields function.

The hash function and HashOptions, since it is not specific to the DeviceDeactivateHash deduplicator, should use the identity fields version.

Or, for complete separation of concerns, you could add a third version, the hash version, either "0" or "1".

So, then the code maps from deduplicator version to hash version to identity fields version.

The DeviceDeactivateHash code looks at its version, maps its version to a hash version and creates the HashOptions with that version. Then, the hash code looks at its version, maps its version to an identity field versions and calls IdentityFields with that identity field version.

Then, you've completely isolated the DeviceDeactivationHash deduplicator from the hash functionality from the identity fields functionality. In practice, we definitely have to do the first step (deduplicator version to hash version) because there will be another duplicator for continuous data sets that is not based off the device id, but the data set id, and doesn't deactivate, but deletes. 

Technically, we don't actually need the hash version and could just use the identity fields version for both hash function and IdentityFields function.

Does that make sense? Let's chat about this tomorrow.